From 903f5a989dacbb99d851dd2bcf7761da3b5abe9b Mon Sep 17 00:00:00 2001
From: Francesco Pham <francesco.pham1@h-partners.com>
Date: Wed, 21 Oct 2026 22:09:12 +0000
Subject: [PATCH] libvpx: keep the x86_64 rtcd generation out of the source
 tree

build.sh generated the x86_64 rtcd headers in the source root and then
removed them, deleting the tracked vpx_scale_rtcd.h that the arm/arm64
builds include (vpx_scale.c: './vpx_scale_rtcd.h' file not found as soon
as an x86_64 product had been built in the same tree). Write the
generated headers and the temporary rtcd config straight into the
output dir instead.

Change-Id: I3331b5b93a05eb33a05f69e97908acce633e1548
Signed-off-by: Francesco Pham <francesco.pham1@h-partners.com>
---
 build.sh | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/build.sh b/build.sh
index d1fdcc988..d7b4c8390 100755
--- a/build.sh
+++ b/build.sh
@@ -69,7 +69,10 @@ if [ "$CPU" == "x86_64" ]; then
       --disable-webm-io \
       --disable-runtime-cpu-detect
 
-  cat > rtcd_config.mk <<'EOF'
+  # generate into the output dir: the tracked vpx_scale_rtcd.h in the source
+  # root is the one the arm builds include
+  RTCD_CONFIG="$OUTPUT_DIR/rtcd_config.mk"
+  cat > "$RTCD_CONFIG" <<'EOF'
 CONFIG_RUNTIME_CPU_DETECT=no
 CONFIG_VP8_DECODER=yes
 CONFIG_VP8_ENCODER=no
@@ -85,22 +88,21 @@ CONFIG_LIBYUV=no
 EOF
 
   perl build/make/rtcd.pl --arch=generic --sym=vp8_rtcd \
-      --config=rtcd_config.mk vp8/common/rtcd_defs.pl > vp8_rtcd.h
+      --config="$RTCD_CONFIG" vp8/common/rtcd_defs.pl > "$OUTPUT_DIR/vp8_rtcd.h"
   perl build/make/rtcd.pl --arch=generic --sym=vp9_rtcd \
-      --config=rtcd_config.mk vp9/common/vp9_rtcd_defs.pl > vp9_rtcd.h
+      --config="$RTCD_CONFIG" vp9/common/vp9_rtcd_defs.pl > "$OUTPUT_DIR/vp9_rtcd.h"
   perl build/make/rtcd.pl --arch=generic --sym=vpx_dsp_rtcd \
-      --config=rtcd_config.mk vpx_dsp/vpx_dsp_rtcd_defs.pl > vpx_dsp_rtcd.h
+      --config="$RTCD_CONFIG" vpx_dsp/vpx_dsp_rtcd_defs.pl > "$OUTPUT_DIR/vpx_dsp_rtcd.h"
   perl build/make/rtcd.pl --arch=generic --sym=vpx_scale_rtcd \
-      --config=rtcd_config.mk vpx_scale/vpx_scale_rtcd.pl > vpx_scale_rtcd.h
+      --config="$RTCD_CONFIG" vpx_scale/vpx_scale_rtcd.pl > "$OUTPUT_DIR/vpx_scale_rtcd.h"
 
   sed -i 's/#define CONFIG_INSTALL_DOCS[[:space:]]*1/#define CONFIG_INSTALL_DOCS 0/' vpx_config.h
   sed -i 's/#define CONFIG_POSTPROC[[:space:]]*1/#define CONFIG_POSTPROC 0/' vpx_config.h
   sed -i 's/#define CONFIG_MULTITHREAD[[:space:]]*0/#define CONFIG_MULTITHREAD 1/' vpx_config.h
   sed -i 's/#define CONFIG_RUNTIME_CPU_DETECT[[:space:]]*1/#define CONFIG_RUNTIME_CPU_DETECT 0/' vpx_config.h
 
-  cp vpx_config.h vp8_rtcd.h vp9_rtcd.h vpx_dsp_rtcd.h vpx_scale_rtcd.h "$OUTPUT_DIR/"
-  rm -f rtcd_config.mk
-  rm -f vpx_config.h vp8_rtcd.h vp9_rtcd.h vpx_dsp_rtcd.h vpx_scale_rtcd.h
+  cp vpx_config.h "$OUTPUT_DIR/"
+  rm -f "$RTCD_CONFIG" vpx_config.h
   exit 0
 fi
 
-- 
2.50.1

